src/components/organisms/DialogReplyInput/DialogReplyInput.tsx

```diff
@@ -0,0 +1,65 @@
+'use client';
+
+import { useState } from 'react';
+import * as Atoms from '@/atoms';
+import * as Core from '@/core';
+import * as Molecules from '@/molecules';
+import * as Organisms from '@/organisms';
+import * as Hooks from '@/hooks';
+
+interface DialogReplyInputProps {
+  postId: string;
+  onSuccess?: () => void;
+}
+
+export function DialogReplyInput({ postId, onSuccess }: DialogReplyInputProps) {
+  const [tags, setTags] = useState<Array<{ id: string; label: string }>>([]);
+  const { replyContent, setReplyContent, handleReplySubmit } = Hooks.usePostReply({ postId, onSuccess });
+  const { ref: containerRef } = Hooks.useElementHeight();
+  const currentUserId = Core.useAuthStore((state) => state.selectCurrentUserPubky());
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter' && !e.shiftKey) {
+      e.preventDefault();
+      void handleReplySubmit();
+    }
+  };
```

Comment:
I noticed we duplicate this function across the codebase a lot, but it looks like @catch-21 made a hook to extract this into a reuseable function here `src/hooks/useEnterSubmit/useEnterSubmit.ts`. So we should use that instead (updating all areas of the code can probably be done in a follow-up task though).